mlflow/utils/validation.py

```diff
@@ -80,7 +82,12 @@ def path_not_unique(name):
 
 def _validate_metric_name(name):
     """Check that `name` is a valid metric name and raise an exception if it isn't."""
-    if name is None or not _VALID_PARAM_AND_METRIC_NAMES.match(name):
+    if name is None:
```

Comment:
I split this out since the Exception thrown for a `None` key name threw the same for an invalid name string and I could see how a user might find this confusing. Is this ok or should I revert this?